resfit/fit_resonator/fit_S_data.py

```diff
@@ -1240,11 +1241,10 @@ def fit_resonator(filename: str,Method,normalize: int,dir: str = None, data_arra
         except:
             print(">Failed to plot CPZM fit for data")
             quit()
-    filename_without_period = ''
-    for i in filename:
-        if i != '.':
-            filename_without_period = filename_without_period + i
-    fig.savefig(output_path+filename_without_period+'_'+Method.method+'_fit.png')
+    if filename.endswith('.csv'):
+        filename = filename[:-4]
+    filename = filename.replace('.','p')
+    fig.savefig(output_path+Method.method+'_'+filename+'.png')
```

Comment:
What's the desired filename from this?

Reply:
Something like "DCM_E06_1_7p308GHz_HPscan_-35dB_13p6mK.png" where E06_1_7p308GHz_HPscan_-35dB_13p6mK is the name of the VNA data.

Reply:
Can you put all the filenaming functinality into it's own function? That'll give us a bit more flexibility later on.

Reply:
In fit_S_data.py, we have pieces of code for naming the folder and for naming the files. Were you thinking both would go in the same function/class? I'm having trouble visualizing how it would be organized.